samples/subsys/usb/mass/sample.usb.mass_sdhc_fatfs fails to build for mimxrt595_evk/mimxrt595s/cm33

**Describe the bug**
samples/subsys/usb/mass/sample.usb.mass_sdhc_fatfs fails to build for mimxrt595_evk/mimxrt595s/cm33
in weekly CI

**To Reproduce**
Steps to reproduce the behavior:
1. twister -p mimxrt595_evk/mimxrt595s/cm33  -T ./samples/subsys/usb/mass/  -s sample.usb.mass_sdhc_fatfs 
4. See error

**Expected behavior**
No build errors, or platform filtered if test not supported

**Impact**
Failure in weekly CI

**Logs and console output**
```
CMake Warning at zephyr/subsys/usb/device/CMakeLists.txt:22 (message):
  CONFIG_USB_DEVICE_VID has default value 0x2FE3.

  This value is only for testing and MUST be configured for USB products.


Load components for MIMXRT595S_cm33:
driver_common component is included.
driver_reset component is included.
device_CMSIS component is included.
CMSIS_Include_core_cm component is included.
device_system component is included.
driver_lpc_gpio component is included.
driver_pint component is included.
driver_inputmux component is included.
driver_inputmux_connections component is included.
driver_ostimer component is included.
driver_flexcomm component is included.
driver_flexcomm_usart component is included.
driver_usdhc component is included.
driver_cache_cache64 component is included.
driver_lpc_iopctl component is included.
middleware_usb_phy component is included.
middleware_usb_common_header component is included.
middleware_usb_device_ip3511fs component is included.
middleware_usb_device_common_header component is included.
-- Using ccache: /usr/bin/ccache
CMake Error at CMakeLists.txt:10 (message):
  No disk access settings detected.


-- Configuring incomplete, errors occurred!
``` 

**Environment (please complete the following information):**

 - OS: Linux Ubuntu 22.04 CI
 - Toolchain: Zephyr SDK 0.16.8
 - Commit SHA or Version used: Main as of now 988e4cf77094fcabaf9a10ad94102d1e4522d0f6

**Additional context**
Introduced by 6bd77c848cafe1b5c2cb2bd60f9b7f369ee80cc5
#75753

--- a/samples/subsys/usb/mass/CMakeLists.txt
+++ b/samples/subsys/usb/mass/CMakeLists.txt
@@ -4,12 +4,6 @@ cmake_minimum_required(VERSION 3.20.0)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(mass)
 
-if((NOT CONFIG_DISK_DRIVER_FLASH) AND
-   (NOT CONFIG_DISK_DRIVER_RAM) AND
-   (NOT CONFIG_DISK_DRIVER_SDMMC))
-	message( FATAL_ERROR "No disk access settings detected." )
-endif()
-
 include(${ZEPHYR_BASE}/samples/subsys/usb/common/common.cmake)
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})

--- a/samples/subsys/usb/mass/sample.yaml
+++ b/samples/subsys/usb/mass/sample.yaml
@@ -117,6 +117,7 @@ tests:
         - "The device is put in USB mass storage mode."
   sample.usb.mass_sdhc_fatfs:
     min_ram: 32
+    filter: dt_compat_enabled("zephyr,sdmmc-disk")
     modules:
       - fatfs
     depends_on:
@@ -138,6 +139,7 @@ tests:
         - "The device is put in USB mass storage mode."
   sample.usb_device_next.mass_sdhc_fatfs:
     min_ram: 32
+    filter: dt_compat_enabled("zephyr,sdmmc-disk")
     modules:
       - fatfs
     depends_on:

--- a/samples/subsys/usb/mass/src/main.c
+++ b/samples/subsys/usb/mass/src/main.c
@@ -30,6 +30,12 @@ LOG_MODULE_REGISTER(main);
 FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(storage);
 #endif
 
+#if !defined(CONFIG_DISK_DRIVER_FLASH) && \
+	!defined(CONFIG_DISK_DRIVER_RAM) && \
+	!defined(CONFIG_DISK_DRIVER_SDMMC)
+#error No supported disk driver enabled
+#endif
+
 #define STORAGE_PARTITION		storage_partition
 #define STORAGE_PARTITION_ID		FIXED_PARTITION_ID(STORAGE_PARTITION)
 

